File: sdk/typescript/examples/basic-usage.ts
/**
 * Basic Usage Example
 * 
 * Run with: npx tsx examples/basic-usage.ts
 */

import { InfinityAssistantClient } from '../src';

async function main() {
  // Initialize client
  const client = new InfinityAssistantClient({
    apiKey: process.env.INFINITY_ASSISTANT_API_KEY || 'ia_your_api_key_here',
  });

  try {
    // Health check
    console.log('Checking API health...');
    const health = await client.health();
    console.log('Health:', health);

    // Send a chat message
    console.log('\nSending chat message...');
    const response = await client.chat({
      message: 'Hello! Can you help me understand TypeScript?',
      mode: 'assist',
    });

    console.log('Response:', response.response);
    console.log('Conversation ID:', response.conversationId);

    // Search knowledge base
    console.log('\nSearching knowledge base...');
    const knowledge = await client.searchKnowledge({
      query: 'TypeScript best practices',
      limit: 5,
    });

    console.log(`Found ${knowledge.total} results`);
    knowledge.results?.forEach((result, index) => {
      console.log(`${index + 1}. ${result.title || result.id}`);
      console.log(`   ${result.content.substring(0, 100)}...`);
    });

  } catch (error: any) {
    console.error('Error:', error.message);
    if (error.statusCode) {
      console.error('Status Code:', error.statusCode);
    }
    if (error.code) {
      console.error('Error Code:', error.code);
    }
  }
}

main();

File: sdk/python/examples/basic_usage.py
"""
Basic Usage Example

Run with: python examples/basic_usage.py
"""

import os
from infinity_assistant import InfinityAssistantClient

def main():
    # Initialize client
    client = InfinityAssistantClient(
        api_key=os.getenv("INFINITY_ASSISTANT_API_KEY", "ia_your_api_key_here")
    )

    try:
        # Health check
        print("Checking API health...")
        health = client.health()
        print("Health:", health)

        # Send a chat message
        print("\nSending chat message...")
        response = client.chat(
            message="Hello! Can you help me understand Python?",
            mode="assist"
        )

        print("Response:", response.get("response"))
        print("Conversation ID:", response.get("conversationId"))

        # Search knowledge base
        print("\nSearching knowledge base...")
        knowledge = client.search_knowledge(
            query="Python best practices",
            limit=5
        )

        print(f"Found {knowledge.get('total', 0)} results")
        for i, result in enumerate(knowledge.get("results", []), 1):
            print(f"{i}. {result.get('title', result.get('id'))}")
            content = result.get("content", "")
            print(f"   {content[:100]}...")

    except Exception as e:
        print(f"Error: {e}")
        if hasattr(e, "status_code"):
            print(f"Status Code: {e.status_code}")
        if hasattr(e, "code"):
            print(f"Error Code: {e.code}")

if __name__ == "__main__":
    main()

